src/interfaces/records/messages/records-upload.ts

```diff
@@ -0,0 +1,516 @@
+    // can't be the upload complete if the message is not a Records Upload
+    if (message.descriptor.interface !== DwnInterfaceName.Records ||
+        message.descriptor.method !== DwnMethodName.Upload) {
+      return false;
+    }
+
+    if (message.descriptor.state !== DwnStateName.Complete) {
+      return false;
+    }
+
+    const recordsUploadCompleteMessage = message as RecordsUploadCompleteMessage;
+    const author = Message.getAuthor(message);
+    const entryId = await RecordsUpload.getEntryId(author, recordsUploadCompleteMessage.descriptor);
+    return entryId === recordsUploadCompleteMessage.recordId;
+  }
+
+  /**
+   * Creates the `attestation` property of a RecordsUpload message if given signature inputs; returns `undefined` otherwise.
+   */
+  private static async createAttestation(descriptorCid: string, signatureInputs?: SignatureInput[]): Promise<GeneralJws | undefined> {
+    if (signatureInputs === undefined || signatureInputs.length === 0) {
+      return undefined;
+    }
+
+    const attestationPayload: RecordsAttestationPayload = { descriptorCid };
+    const attestationPayloadBytes = Encoder.objectToBytes(attestationPayload);
+
+    const signer = await GeneralJwsSigner.create(attestationPayloadBytes, signatureInputs);
+    return signer.getJws();
+  }
+
+  /**
+   * Creates the `authorization` property of a RecordsUpload message.
+   */
+  private static async createAuthorization(
+    recordId: string,
+    descriptorCid: string,
+    attestation: GeneralJws | undefined,
+    signatureInput: SignatureInput
+  ): Promise<GeneralJws> {
+    const authorizationPayload: RecordsAuthorizationPayload = {
+      recordId,
+      descriptorCid
+    };
+
+    const attestationCid = attestation ? await computeCid(attestation) : undefined;
+
+    if (attestationCid !== undefined) { authorizationPayload.attestationCid = attestationCid; } // assign `attestationCid` only if it is defined
+
+    const authorizationPayloadBytes = Encoder.objectToBytes(authorizationPayload);
+
+    const signer = await GeneralJwsSigner.create(authorizationPayloadBytes, [signatureInput]);
+    return signer.getJws();
+  }
+
+  /**
+   * Gets the upload start from the given list or record upload.
+   */
+  public static async getUploadStart(messages: BaseMessage[]): Promise<RecordsUploadStartMessage>{
+    for (const message of messages) {
+      if (await RecordsUpload.isUploadStart(message)) {
+        return message as RecordsUploadStartMessage;
+      }
+    }
+
+    throw new Error(`upload start is not found`);
+  }
+
+  /**
+   * Gets the upload complete from the given list or record upload.
+   */
+  public static async getUploadComplete(messages: BaseMessage[]): Promise<RecordsUploadCompleteMessage>{
+    for (const message of messages) {
+      if (await RecordsUpload.isUploadComplete(message)) {
+        return message as RecordsUploadCompleteMessage;
+      }
+    }
+
+    throw new Error(`upload complete is not found`);
```

Comment:
Strikes me as a bit odd to throw on what feels like valid input in an unexceptional situation. Esp since the only usage of this function swallows the error. Thoughts on returning `undefined`?